pkg/operation/botanist/monitoring.go

```diff
@@ -60,7 +61,26 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 		usersDashboards     = strings.Builder{}
 	)
 
-	// Find extensions provider-specific monitoring configuration
+	// Fetch component-specific monitoring configuration
+	for _, component := range []component.MonitoringComponent{
+		b.Shoot.Components.ControlPlane.KubeScheduler,
+	} {
+		scrapeConfig, err := component.ScrapeConfig()
+		if err != nil {
+			return err
+		}
+		scrapeConfigs.WriteString(fmt.Sprintln(scrapeConfig))
+
+		alertingRule, err := component.AlertingRules()
+		if err != nil {
+			return err
+		}
+		for filename, rule := range alertingRule {
+			alertingRules.WriteString(fmt.Sprintf("%s: |\n  %s\n", filename, strings.ReplaceAll(rule, "\n", "\n  ")))
```

Comment:
Now, alerting rules and scrape config of the scheduler are passed to values under `extensions` which seems a bit off. 
Can you use separate values for that? I would also vote for using `map[string]string` for the components' `alertingRules` and merging maps instead of appending YAML-formatted strings. `map[string]string` can be passed to the values directly.

Reply:
> Now, alerting rules and scrape config of the scheduler are passed to values under extensions which seems a bit off.

I don't see any benefit of duplicating this part in the Prometheus Helm chart. From the chart's API point of view these things are just additional configurations, no matter where they are coming from. I can rename the value key from `extensions` to `additional{...}` to make that more clear in the chart itself, but I don't see any benefit of duplicating the rendering logic.

> I would also vote for using map[string]string for the components' alertingRules and merging maps instead of appending YAML-formatted strings. map[string]string can be passed to the values directly.

Can we do this as part of the Prometheus deployment refactoring? This PR is about the kube-scheduler and I don't want to mix too many things at once. I agree that the string formatting is odd, but we might have to keep it anyway (at least for a while) due to the current monitoring/logging extension contract. I'd vote for cleaning this up in a dedicated improvement track. For newly introduced components that implement the `MonitoringComponent` interface it is easy/clear enough, or?

Reply:
Yes, agreed. Let's do it in the prometheus component refactoring.